app/signIn/page.tsx

```diff
@@ -1,23 +1,37 @@
 'use client';
 import {useForm} from "react-hook-form";
 import pb from "@/app/lib/pocketbase"
-import {useState} from "react";
+import {useContext, useEffect, useState} from "react";
 import {randomInt} from "node:crypto";
-
+import UserContext from "@/app/userContext"
 
 export default function SignIn() {
     const {register, handleSubmit, reset} = useForm();
     const [isLoading, setIsLoading] = useState<boolean>(false);
     const [isLogged, setIsLogged] = useState<boolean>(false);
-    const [dummy ,setDummy] = useState(0);
+    const [dummy, setDummy] = useState(0);
+    const [user, setUser] = useContext(UserContext);
+
+    useEffect(() => {
```

Comment:
could extract this into a more modular hook like:

`const isLoggedIn = () => {
   const [isLoggedIn, setIsLoggedIn] = useState<boolean>(false);
   
   useEffect(() => {
        const storedUser = localStorage.getItem('user');
        if (storedUser) {
            setUser(storedUser);
            setIsLoggedIn(true);
        }
    }, []);
    
    return {
      isLoggedIn
   }   
`